### PR TITLE
Used trusted publisher for PyPi

### PR DESCRIPTION
**Pull request by woodsp-ibm**
### Summary

This PR follows Qiskit/rustworkx#1001 to update the release CI workflow to use PyPI's trusted publisher mechanism.

### CODE REVIEW

**Review by mtreinish — APPROVED** [review]
This LGTM, I'm planning to hopefully migrate the pypi account on Wednesday, so as long as you aren't planning any releases before then we can merge this whenever (it only will work after the migration though).

**Review by mrossinek — APPROVED** [review]

**Pull request merged**
